pbjam/priors.py

```diff
@@ -84,10 +84,10 @@ def _downsample_prior_check(self, idx, KDEsize):
             Number of targets to include in the KDE estimation.
 
         """
-        if len(idx) < KDEsize:
-            warnings.warn(f'Sample for estimating KDE is less than the request {KDEsize}.')
-            KDEsize = len(idx)
-
+        if len(idx[idx==1]) < KDEsize:
```

Comment:
I don't get this change - what is happening here? idx[idx==1] :). Could you add a comment for future ref?

Reply:
idx is a boolean array of shape equal to the whole list of prior_data.csv, with 1's for targets that are within range in numax of the input target, and 0's otherwise. 

KDEsize is the requested number of targets to compute the KDE. 

If the number of 1's in idx is less than KDEsize (100 by default), it'll show a warning and sets KDEsize to the number of targets in range. This happens after having tried to scale up the range to 20sigma. 

KDEsize and idx are used to draw a random sample from the prior_data.csv dataframe, where idx is used as weights and KDEsize is the number to draw. pandas will throw an error if KDEsize is greater than the number of non-zero elements in the weights array, so they must at least be equal.

If there's a nicer way of doing this, feel free to add it.

Reply:
If I were in charge I would add the following as a comment:

> idx is a boolean array of shape equal to the whole list of prior_data.csv, with 1's for targets that are within range in numax of the input target, and 0's otherwise.

:)

Reply:
Sure, I'll add that, but I think I'll put that in the docstring.